R/ddl.R

```diff
@@ -0,0 +1,210 @@
+#' DDL object
+#'
+#' Object to execute custom DDL code in the shiny session
+#'
+#' @param expr (`expression`)\cr
+#'  Syntatically valid R code to be executed in the shiny session.
+#'  shouldn't be specified when `code` is specified.
+#'
+#' @param code (`character`, `language`)\cr
+#'  Object containing code to be evaluated to load data. Shouldn't be specified when `expr`
+#'  is specified.
+#'
+#'
+#' @param ui (`shiny.tag`)\cr
+#'   `shiny` ui module containing inputs which `id` correspond to the
+#'   args in the `code`.
+#'
+#' @param server (`function`)\cr
+#'   `shiny` server module returning data. This server should execute
+#'   `code` and return a reactive data containing necessary data. To handle
+#'   evaluation and code masking process it is recommended to use `ddl_run`.
+#'   Package provides universal `username_password_server` which
+#'   runs `ddl_run` function, which returns `teal_data` object.
+#'   Details in the the example
+#'
+#' @param input_mask (`list` named)\cr
```

Comment:
This name could be more precise. Something like: `obfuscated_inputs` or `hidden_inputs` or `replaced_inputs` better conveys the intentions, I think. Masking has a very specific meaning in either statistical programming or programming in general, and it has little to do with making sure that something is not visible.

Reply:
Thanks @kpagacz . Agree and not. "Masking" in R has a specific meaning which is a synonym of replacement or overriding. For example overriding function when loading namespaces is called "masking" ("function x has been masked"). [data-masking](https://search.r-project.org/CRAN/refmans/rlang/html/topic-data-mask.html) is apparently a thing.

Reply:
To be fair, I was thinking in colloquial terms when I first proposed `mask` but it does express the meaning quite well.

Reply:
I subconsciously understood the purpose of `input_mask` after the first time I saw it. I would eventually propose `input_masking` as an alternative name, but it's longer.